src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -84,68 +84,66 @@ public TimeOnly(long ticks)
                 throw new ArgumentOutOfRangeException(nameof(ticks), SR.ArgumentOutOfRange_TimeOnlyBadTicks);
             }
 
-            _ticks = ticks;
+            _ticks = (ulong)ticks;
         }
 
         // exist to bypass the check in the public constructor.
-        internal TimeOnly(ulong ticks) => _ticks = (long)ticks;
+        internal TimeOnly(ulong ticks) => _ticks = ticks;
 
         /// <summary>
         /// Gets the hour component of the time represented by this instance.
         /// </summary>
-        public int Hour => new TimeSpan(_ticks).Hours;
+        public int Hour => (int)(_ticks / TimeSpan.TicksPerHour);
 
         /// <summary>
         /// Gets the minute component of the time represented by this instance.
         /// </summary>
-        public int Minute => new TimeSpan(_ticks).Minutes;
+        public int Minute => (int)((uint)(_ticks / TimeSpan.TicksPerMinute) % 60);
 
         /// <summary>
         /// Gets the second component of the time represented by this instance.
         /// </summary>
-        public int Second => new TimeSpan(_ticks).Seconds;
+        public int Second => (int)((uint)(_ticks / TimeSpan.TicksPerSecond) % 60);
```

Comment:
>  public int Second => (int)((uint)(_ticks / TimeSpan.TicksPerSecond) % 60); [](http://example.com/codeflow?start=6&length=76)

I am wondering in general if these are worth optimizing? Having the calculation in one place in TimeSpan is better IMO.

Reply:
`TimeSpan` supports large ranges, including negative values. `TimeOnly` values are more constrained.

Reply:
Yes, I understand that. I was wondering if it is worth the minor perf gain to repeat the calculation here.